src/m365/aad/commands/administrativeunit/administrativeunit-roleassignment-add.ts

```diff
@@ -0,0 +1,137 @@
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  administrativeUnitId?: string;
+  administrativeUnitName?: string;
+  roleDefinitionId?: string;
+  roleDefinitionName?: string;
+  userId?: string;
+  userName?: string;
+}
+
+class AadAdministrativeUnitRoleAssignmentAddCommand extends GraphCommand {
+  public get name(): string {
+    return commands.ADMINISTRATIVEUNIT_ROLEASSIGNMENT_ADD;
+  }
+
+  public get description(): string {
+    return 'Assigns a Microsoft Entra role with administrative unit scope to a user';
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+    this.#initOptionSets();
+    this.#initTypes();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        administrativeUnitId: typeof args.options.administrativeUnitId !== 'undefined',
+        administrativeUnitName: typeof args.options.administrativeUnitName !== 'undefined',
+        roleDefinitionId: typeof args.options.roleDefinitionId !== 'undefined',
+        roleDefinitionName: typeof args.options.roleDefinitionName !== 'undefined',
+        userId: typeof args.options.userId !== 'undefined',
+        userName: typeof args.options.userName !== 'undefined'
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-i, --administrativeUnitId [administrativeUnitId]'
+      },
+      {
+        option: '-n, --administrativeUnitName [administrativeUnitName]'
+      },
+      {
+        option: '--roleDefinitionId [roleDefinitionId]'
+      },
+      {
+        option: '--roleDefinitionName [roleDefinitionName]'
+      },
+      {
+        option: '--userId [userId]'
+      },
+      {
+        option: '--userName [userName]'
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.administrativeUnitId && !validation.isValidGuid(args.options.administrativeUnitId as string)) {
+          return `${args.options.administrativeUnitId} is not a valid GUID`;
+        }
+
+        if (args.options.roleDefinitionId && !validation.isValidGuid(args.options.roleDefinitionId as string)) {
+          return `${args.options.roleDefinitionId} is not a valid GUID`;
+        }
+
+        if (args.options.userId && !validation.isValidGuid(args.options.userId as string)) {
+          return `${args.options.userId} is not a valid GUID`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push({ options: ['administrativeUnitId', 'administrativeUnitName'] });
+    this.optionSets.push({ options: ['roleDefinitionId', 'roleDefinitionName'] });
+    this.optionSets.push({ options: ['userId', 'userName'] });
+  }
+
+  #initTypes(): void {
+    this.types.string.push('administrativeUnitName', 'roleName', 'userName');
```

Comment:
Is there a specific reason we explicitly define these values as strings?

Reply:
Without .string. it throws the error: `Property 'push' does not exist on type 'CommandTypes'`

Reply:
I mean, why we use `#initTypes` here in the first place

Reply:
Not needed. Removed.